js-packages/@prestojs/rest/src/Endpoint.ts

```diff
@@ -509,54 +643,107 @@ export default class Endpoint<ReturnT = any> {
             options = paginator.getRequestInit(restOptions);
         }
         const { urlArgs = {}, query, ...init } = options;
```

Comment:
The paginator is causing `options` to not be fully preserved for `execute()` later on; I could maybe see a use case if you have a limit on how many things you can fetch and want to chain say up to 10 pages of data at a time, but make it look to the caller as though you simply had a very large page size